README.md

```diff
@@ -1,28 +1,69 @@
 # cached-module-loader
 
-Highly experimental bundler and loader of Node.js modules.
+> Caching bundler and loader of Node.js modules using `vm` module's `cachedData` feature implemented since Node.js v6.
 
-**Don't use this.**
+☣ **Highly experimental, don't use it.**
 
-## Usage ;-)
+## Installation
+
+```
+$ npm install --save cached-module-loader
+```
+
+## Usage
+
+Generate a cached bundle:
 
 ```js
-const cml = require('cached-module-loader')
+const { join } = require('path')
+const { writeFileSync } = require('fs')
+const cachedModuleLoader = require('cached-module-loader')
 
-cml.bundleDependencies('babel-core').then(bundle => {
-  const babel = cml.load(require.resolve('babel-core'), module, bundle)
-  // And now use babel!
+cachedModuleLoader.bundleDependencies('babel-core').then(bundle => {
+  writeFileSync(join(__dirname, '.cached-data.bin'), cachedData)
+  writeFileSync(join(__dirname, '.cached-code.js'), code)
 })
 ```
 
-## Performance tests
+Read a bundle and include module:
 
-Clone this repository, then:
+```js
+const { join } = require('path')
+const { readFileSync } = require('fs')
+const cachedModuleLoader = require('cached-module-loader')
 
-```sh
-npm install
-node perf-tests/create-cache-data.js
+const cachedData = readFileSync(join(__dirname, '.cached-data.bin'))
+const code = readFileSync(join(__dirname, '.cached-code.js'))
 
-time node perf-tests/require-with-cached-data.js
-time node perf-tests/raw-require.js
+const babel = loadInThisContext(require.resolve('babel-core'), module, {
+  moduleId: 'babel-core',
+  cachedData,
+  code
+})
 ```
+
+## How it works
+
+First, `cached-module-loader` records every dependency of the requested module and
+rewrites their paths to be absolute. Then all dependencies get concatenated into
+one bundle (kind of like `browserify` works). Each module is also wrapped into a
+customized module wrapper, with the replaced `require()` function. The replacement
+is modified to load modules from that concatenated bundle.
```

Comment:
That's a good start, thanks. I think it's worth mentioning that we're only bundling `.js` dependencies. The `require()` behavior is a little different from how you described it too, e.g. the standard module wrapper is used, it just receives a differently behaving function.

How about this?

> First `cached-module-loader` records every `.js` dependency of the requested module. These dependencies (and the module itself) get concatenated into one bundle. `require()` calls are rewritten to use absolute paths. The dependencies are wrapped in Node.js' standard module wrapper, though when loading modules a modified `require()` function is passed. This function can load modules from the bundle rather than from disk.